### PR TITLE
Write analysis for concept exercise `blorkemon-cards`

### LINKED ISSUE #42: Write analysis for concept exercise `blorkemon-cards`
[From the design file](https://github.com/exercism/elm/blob/main/exercises/concept/blorkemon-cards/.meta/design.md)

Make sure that:

- (essential) `sortByMonsterName` uses `List.sortBy`
- (essential) `sortByCoolness` uses `List.sortBy`
- (essential) `expectedWinner` uses `compareShinyPower`
- (essential) `expectedWinner` uses `case`

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #42.

I didn't follow exactly the design document, I added one to check that `maxPower` uses `max` (to my surprise, someone didn't) and I removed `sortByCoolness` uses `List.sortBy` because the hints also present `List.sortWith` as a possibility.

~~I'm too sleepy for doing the comment PR, so that will have to wait for tomorrow, I think.~~
[Sister PR here](https://github.com/exercism/website-copy/pull/2229).

The CI broke because the brand new version of elm-test requires `elm-explorations/test` 2.0, so I fixed it for now, I will do a dependency update later.

### CODE REVIEW

**Comment by mpizenberg** [comment]
> The CI broke because the brand new version of elm-test requires elm-explorations/test 2.0, so I fixed it for now, I will do a dependency update later.

Regarding this, it would be nice if you could add an `elm-tooling.json` as in the elm repo https://github.com/exercism/elm/blob/main/elm-tooling.json
And with that, you can use the elm-tooling-action github action that installs the specified tools much more efficiently than npm. (cf https://github.com/exercism/elm/blob/main/.github/workflows/build.yml#L25)
Benefits is you know exactly the version of the tools used both locally and in CI. Elm-test is not available via elm-tooling though so you still need to install it via npm, or switch to elm-test-rs which can be installed via elm-tooling.

**Comment by jiegillet** [comment]
Yep, sounds good.
I seem to remember elm-test-rs didn't work here for some reason but I'll investigate. Probably a different PR, if it's OK I'll ping you to review it!

File: tests/Exercise/BlorkemonCardsTest.elm
Comment (ceddlyburge) [inline tests/Exercise/BlorkemonCardsTest.elm:205]: This looks better than the way I did it, as it doesn't rely on the position of the error in the string. I'll have to try and do it like this next time ...

**Review by ceddlyburge — APPROVED** [review]

File: .github/workflows/elm_tests.yml
Comment (jiegillet) [reply .github/workflows/elm_tests.yml:22]: Agreed 👍

File: src/Analyzer.elm
Comment (jiegillet) [reply src/Analyzer.elm:199]: Isn't it? I think I stumbled upon good design here ;)

File: src/Exercise/BlorkemonCards.elm
Comment (jiegillet) [reply src/Exercise/BlorkemonCards.elm:29]: Actually to be honest, "maxPower doesn't use max" is not used anywhere, it's only there for the sake of humans explaining when this comment would pop up.
It makes tests and rules a little bit more readable, that's all. Maybe it's not worth it...

**Comment by jiegillet** [comment]
Thank you for the review @ceddlyburge 

@exercism/maintainers-admin can I get an approval please? 💛

File: src/Exercise/BlorkemonCards.elm
Comment (ceddlyburge) [reply src/Exercise/BlorkemonCards.elm:29]: I think I would prefer to remove these in that case, but probably wants a longer discussions and definitely a different pr. Maybe we could just make the elm.blorkemon-cards.use_max more descriptive?

Comment (jiegillet) [reply src/Exercise/BlorkemonCards.elm:29]: Yes, maybe you could open an issue?
We cannot touch the `elm.blorkemon-cards.use_max` field, because it follows Exercism specs. We could use mores descriptive file names, maybe, but I'm afraid of adding too much complexity there.

**Pull request merged**
